src/infrastructure/httpServer/common/HttpError.ts

```diff
@@ -1,37 +1,53 @@
 import { DocumentationLink } from "../../../DocumentationLink";
 
+export interface HttpErrorJSON {
+    code: string;
+    message: string;
+    id: string;
+    time: string;
+    docs: string;
+    stacktrace?: string[];
+    details?: string;
+}
+
 export class HttpError {
     public readonly docs: string;
+    public readonly id: string;
+    public readonly time: string;
 
-    public constructor(
-        public readonly code: string,
-        public readonly message: string,
-        public readonly id: string = HttpErrorId.create(),
-        public readonly time: string = new Date().toISOString()
-    ) {
+    private constructor(public readonly code: string, public readonly message: string, public readonly stacktrace?: string[], public readonly details?: string) {
         this.id = HttpErrorId.create();
         this.time = new Date().toISOString();
-        this.code = code;
-        this.message = message;
 
         this.docs = DocumentationLink.integrate__errorCodes(code);
     }
 
-    public static forDev(code: string, message: string, stacktrace: string[], details: string): HttpErrorDev {
-        return new HttpErrorDev(code, message, stacktrace, details);
+    public toJSON(verbose?: "verbose"): HttpErrorJSON {
+        const json: HttpErrorJSON = {
+            code: this.code,
+            message: this.message,
+            id: this.id,
+            time: this.time,
+            docs: this.docs
+        };
+
+        if (verbose) {
+            json.stacktrace = this.stacktrace;
+            json.details = this.details;
+        }
+
+        return json;
+    }
+
+    public static forDev(code: string, message: string, stacktrace: string[], details: string): HttpError {
```

Comment:
Wir brauchen doch jetzt nicht mehr forDev und forProd, oder?

Reply:
IMO können wir es umbenennen, z.B. in with stacktrace oder so.. oder wir benutzen den Connector.

Reply:
Wir prüfen doch jetzt an anderer Stelle, ob wir einen Stacktrace anhängen oder nicht. D.h. der Stacktrace kann immer in den HttpError reingegeben werden